### DISCUSSION TITLE
PLUMED patch for GROMACS modular simulator

### DISCUSSION THREAD

**Discussion by dmandelli** [plumed/plumed2#1039] [General]
Dear developers,

We have implemented a new feature in GROMACS's modular simulator that will benefit from PLUMED (to perform metadynamics) and we would greatly appreciate any kind of guidance to speed up the implementation of a patch.

Best would be to set up a meeting with the PhD student who's working on it.

Feel free to contact me.

Thanks in advance!

Best regards,
Davide

**Comment by GiovanniBussi** [comment]
Hi Davide,

on our side we would be happy if anyone wants to contribute or help maintaining a patch.

Regarding the possibility to use the modular simulator, I think the best idea would be to use the native GROMACS API for plugins. We were already discussing the possibility to work on it (see [here](https://gitlab.com/gromacs/gromacs/-/issues/4939)), and we already have some idea of what to do (and not to do). In particular, we would avoid having another patch, and at the same time avoid including the full PLUMED codebase in GROMACS. Rather, we would make it possible to load the PLUMED library at run time. Likely, Daniele (@Iximiel ) will work on the later in spring/early summer, because we have other priorities at this time.

If you want to help in this direction I think it would be very appreciated on our side, so let's keep in touch.

Giovanni

**Reply by mabraham** [reply]
@dmandelli Note that the imminent GROMACS 2025.0 release (available in beta form) has integrated support for PLUMED, so there may be a solution space for you to explore now!

**Reply by dmandelli** [reply]
Dear Mark. Thanks for the info! We eventually moved our code to the legacy code of GROMACS and then modified the existing plumed patch. This has been solved.
